src/Raven.Server/Documents/Sharding/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForDeleteOngoingTask.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Server.Documents.Handlers.Processors.OngoingTasks;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.OngoingTasks
+{
+    internal class ShardedOngoingTasksHandlerProcessorForDeleteOngoingTask : AbstractOngoingTasksHandlerProcessorForDeleteOngoingTask<ShardedDatabaseRequestHandler>
+    {
+        public ShardedOngoingTasksHandlerProcessorForDeleteOngoingTask([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override string GetDatabaseName()
+        {
+            return RequestHandler.DatabaseContext.DatabaseName;
+        }
+
+        protected override async ValueTask WaitForIndexNotificationAsync(long index)
+        {
+            await RequestHandler.DatabaseContext.Cluster.WaitForExecutionOnAllNodesAsync(index);
+        }
+
+        protected override async ValueTask RaiseNotificationForSubscriptionTaskRemoval()
+        {
+            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Aviv, DevelopmentHelper.Severity.Normal, "how should we handle other nodes? ");
+            var tasks = RequestHandler.ServerStore.DatabasesLandlord.TryGetOrCreateShardedResourcesStore(RequestHandler.DatabaseContext.DatabaseName);
+            foreach (var task in tasks)
+            {
+                var db = await task;
+                db.SubscriptionStorage.RaiseNotificationForTaskRemoved(TaskName);
```

Comment:
What if this isn't a subscription?

Reply:
if it isn't a subscription we won't reach this code : https://github.com/ravendb/ravendb/pull/14132/files#diff-de95299634152336d17647bf127e1e89995d74938a8350885dcdd0a4920059f3R34-R40